scrape/scrape.go

```diff
@@ -1314,6 +1316,17 @@ loop:
 			if err == storage.ErrNotFound {
 				ok = false
 			}
+
+			// If the appending of sample errored, no point in trying for exemplars.
+			if hasExemplar := p.Exemplar(&e); ok && hasExemplar {
+				if !e.HasTs {
+					e.Ts = t
+				}
+				exemplarErr := app.AddExemplarFast(ce.ref, e)
+				if exemplarErr == storage.ErrNotFound {
```

Comment:
Should we not break the loop otherwise? (should be captured in `err` before breaking I think)